pulsar-broker/src/test/java/org/apache/pulsar/broker/qos/AsyncTokenBucketTest.java

```diff
@@ -48,6 +49,29 @@ private void incrementMillis(long millis) {
         manualClockSource.addAndGet(TimeUnit.MILLISECONDS.toNanos(millis));
     }
 
+    @Test
+    void testAsyncTokenWithMultiCall() throws Exception {
+        int rate = 2000;
+        int resolutionTimeNano = 8;
+        asyncTokenBucket = AsyncTokenBucket.builder().rate(rate).ratePeriodNanos(TimeUnit.SECONDS.toNanos(1)).clock(
+                new DefaultMonotonicSnapshotClock(TimeUnit.MILLISECONDS.toNanos(resolutionTimeNano), System::nanoTime))
+                .build();
+
+        for (int i = 0; i < (1000); i++) {
+            for (int j = 0; j < (1000); j++) {
+                long token = asyncTokenBucket.getTokens();
+                if (token < 0) {
+                    // sleep to allow add new more tokens
+                    Thread.sleep(resolutionTimeNano * 5);
+                    assertTrue(asyncTokenBucket.getTokens() > 0);
```

Comment:
This is an incorrect assumption that tokens would always be positive in the AsyncTokenBucket solution.